lib/trino-hive-formats/src/main/java/io/trino/hive/formats/line/grok/Grok.java

```diff
@@ -623,12 +622,28 @@ public void compile(String pattern, boolean namedOnly)
         }
 
         if (namedRegex.isEmpty()) {
-            throw new GrokException("Pattern not fount");
+            throw new GrokException("Pattern not found");
         }
         // Compile the regex
         compiledNamedRegex = Pattern.compile(namedRegex);
     }
 
+    public String removeUnderscores(String namedRegex)
+    {
+        // Pattern.compile() does not support underscores in named regex groups so need to remove all of them
+        Pattern groupPattern = Pattern.compile("\\(\\?<([^>]+)>");
```

Comment:
Minor: lift this into a `private static final` constant instead of compiling it on each call